libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/GuidanceViewImageProvider.kt

```diff
@@ -0,0 +1,116 @@
+package com.mapbox.navigation.ui.instruction
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import com.mapbox.api.directions.v5.models.BannerComponents
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.navigation.utils.internal.ifNonNull
+import com.squareup.picasso.OkHttp3Downloader
+import com.squareup.picasso.Picasso
+import com.squareup.picasso.Target
+import java.lang.Exception
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+
+/**
+ * The class serves as a medium to emit bitmaps for the respective guidance view URL embedded in
+ * [BannerInstructions]
+ * @constructor
+ */
+class GuidanceViewImageProvider() {
+
+    companion object {
+        private val USER_AGENT_KEY = "User-Agent"
+        private val USER_AGENT_VALUE = "MapboxJava/"
+    }
+
+    /**
+     * This is added a solution to bypass the issue with Picasso where [com.squareup.picasso.Target]
+     * get's garbage collected and guidance view is hence not rendered. Found the solution here
+     * https://stackoverflow.com/questions/24180805/onbitmaploaded-of-target-object-not-called-on-first-load#answers
+     */
+    private val targets: MutableList<Target> = mutableListOf()
+    private val okHttpClient = OkHttpClient.Builder().addInterceptor { chain: Interceptor.Chain ->
+        chain.proceed(
+            chain.request().newBuilder().addHeader(USER_AGENT_KEY, USER_AGENT_VALUE).build()
+        )
+    }.build()
+
+    /**
+     * The API reads the bannerInstruction and renders a guidance view if one is available
+     * @param bannerInstructions [BannerInstructions]
+     * @param context [Context]
+     * @param callback [OnGuidanceImageDownload] Callback that is triggered based on appropriate state of image downloading
+     */
+    fun renderGuidanceView(
+        bannerInstructions: BannerInstructions,
+        context: Context,
+        callback: OnGuidanceImageDownload
+    ) {
+        val target = object : Target {
+            override fun onPrepareLoad(placeHolderDrawable: Drawable?) {
+            }
+
+            override fun onBitmapFailed(e: Exception?, errorDrawable: Drawable?) {
+                targets.clear()
```

Comment:
we should only remove _this_ target instance, not clear all of them. Otherwise, if the first request doesn't return before we schedule the second one, we can run into issues.

Reply:
As explained below, the list of `target` can never have more than 1 target
cc @LukasPaczos